google-cloud-talent-v4/AUTHENTICATION.md

```diff
@@ -0,0 +1,169 @@
+# Authentication
+
+In general, the google-cloud-talent-v4 library uses
+[Service Account](https://cloud.google.com/iam/docs/creating-managing-service-accounts)
+credentials to connect to Google Cloud services. When running within
+[Google Cloud Platform environments](#google-cloud-platform-environments) the
+credentials will be discovered automatically. When running on other
+environments, the Service Account credentials can be specified by providing the
+path to the
+[JSON keyfile](https://cloud.google.com/iam/docs/managing-service-account-keys)
+for the account (or the JSON itself) in
+[environment variables](#environment-variables). Additionally, Cloud SDK
+credentials can also be discovered automatically, but this is only recommended
+during development.
+
+## Quickstart
+
+1. [Create a service account and credentials](#creating-a-service-account).
+2. Set the [environment variable](#environment-variables).
+
+```sh
+export TALENT_CREDENTIALS=path/to/keyfile.json
+```
+
+3. Initialize the client.
+
+```ruby
+require "google/cloud/talent/v4"
+
+client = ::Google::Cloud::Talent::V4::CompanyService::Client.new
```

Comment:
Does it make sense to show `CompanyService` here while the wrapper auth guide shows `client = Google::Cloud::Talent.application_service` and the `V4beta1` guide similarly shows `client = ::Google::Cloud::Talent::V4beta1::ApplicationService::Client.new`?

Reply:
V4 apparently doesn't have an ApplicationService. (One of several services that were dropped between v4beta1 and v4.)

We'll need to switch the wrapper to default to v4 rather than v4beta1. When we do that, the generator will switch the wrapper's guide to use a service that exists in v4, which will probably be CompanyService. (That will be a semver-major change for the wrapper.) Unfortunately, the v1beta1 guide will still show ApplicationService as the example, though.

Reply:
Sounds good, thanks for looking into that!